sorting/binary_insertion_sort.cpp

```diff
@@ -1,146 +1,131 @@
-/**
- * \file
- * \brief [Binary Insertion Sort Algorithm
- * (Insertion Sort)](https://en.wikipedia.org/wiki/Insertion_sort)
- *
- * \details
- * If the cost of comparisons exceeds the cost of swaps, as is the case for
- * example with string keys stored by reference or with human interaction (such
- * as choosing one of a pair displayed side-by-side), then using binary
- * insertion sort may yield better performance. Binary insertion sort employs a
- * binary search to determine the correct location to insert new elements, and
- * therefore performs ⌈log2 n⌉ comparisons in the worst case. When each element
- * in the array is searched for and inserted this is O(n log n). The algorithm
- * as a whole still has a running time of O(n2) on average because of the series
- * * of swaps required for each insertion. However it has several advantages
- * such as
- * 1. Easy to implement
- * 2. For small set of data it is quite efficient
- * 3. More efficient that other Quadratic complexity algorithms like
- *    Selection sort or bubble sort.
- * 4. It is efficient to use it when the cost of comparison is high.
- * 5. It's stable that is it does not change the relative order of
- *    elements with equal keys.
- * 6. It can sort the array or list as it receives.
- *
- * Example execution steps:
- * 1. Suppose initially we have
- * \f{bmatrix}{40 &30 &20 &50 &10\f}
- * 2. We start traversing from 40 till we reach 10
- * when we reach at 30 we find that it is not at it's correct place so we take
- * 30 and place it at a correct position thus the array will become
- * \f{bmatrix}{30 &40 &20 &50 &10\f}
- * 3. In the next iteration we are at 20 we find that this is also misplaced so
- * we place it at the correct sorted position thus the array in this iteration
- * becomes

```

Comment:
this was an unnecessary deletion. This is the format we use for documenting files